shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -1381,6 +1381,16 @@ class JSTestBackend extends JSBackend(allowUnresolvedSymbols = false) {
       case _ => die
     }
 
+    otherStmts.foreach {
+      case fd @ NuFunDef(isLetRec, Var(nme), symNme, _, L(body)) if isLetRec.getOrElse(true) =>
```

Comment:
Why are we using this super weird condition `if isLetRec.getOrElse(true)`? It would deserve some documentation. It seems to somehow filter out specifically non-recursive let bindings, but for some reason include recursive let bindings. 🧐